### PR TITLE
Introduce resource limits for Terraform controller

### PR DESCRIPTION
**Pull request by redsnapper2006**
Fix https://github.com/oam-dev/terraform-controller/issues/244

### CODE REVIEW

**Comment by msftgits** [comment]
[CLA assistant check](https://cla.opensource.microsoft.com/oam-dev/terraform-controller?pullRequest=276) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/oam-dev/terraform-controller?pullRequest=276) before we can accept your contribution.<br/><br/>:x: redsnapper2006 [sign now](https://cla.opensource.microsoft.com/oam-dev/terraform-controller?pullRequest=276)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/oam-dev/terraform-controller?pullRequest=276) it.</sub>

File: controllers/configuration_controller.go [hunk 1]
```diff
@@ -218,6 +219,16 @@ type TFConfigurationMeta struct {
 	TerraformBackendNamespace string
 	BusyboxImage              string
 	GitImage                  string
+
+	// Resources series Variables are for Setting Compute Resources required by this container
```
Comment (zzxwill) [inline controllers/configuration_controller.go:223]: This should be exposed in values.yaml like https://github.com/oam-dev/terraform-controller/blob/master/chart/values.yaml#L8-L10.

File: controllers/configuration_controller.go [hunk 2]
```diff
@@ -409,6 +420,47 @@ func (r *ConfigurationReconciler) preCheck(ctx context.Context, configuration *v
 		meta.GitImage = "alpine/git:latest"
 	}
 
+	meta.ResourcesLimitsCPU = os.Getenv("RESOURCES_LIMITS_CPU")
```
Comment (zzxwill) [inline controllers/configuration_controller.go:423]: These environments should be set in the controller manifests like https://github.com/oam-dev/terraform-controller/blob/master/chart/templates/terraform_controller.yaml#L28-L37.

File: chart/templates/terraform_controller.yaml [hunk 3]
```diff
@@ -35,4 +35,20 @@ spec:
               value: {{ .Values.gitImage}}
             - name: GITHUB_BLOCKED
               value: {{ .Values.githubBlocked }}
+            {{ if .Values.resourceLimitsCPU }}
```
Comment (zzxwill) [inline chart/templates/terraform_controller.yaml:38]:
```suggestion
            {{ if .Values.resources.limits.cpu }}
```
Will it be better to follow the style like https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-requests-and-limits-of-pod-and-container?

Comment (redsnapper2006) [reply chart/templates/terraform_controller.yaml:38]: correct, that is good

File: controllers/configuration_controller_test.go [hunk 4]
```diff
@@ -557,10 +749,149 @@ func TestPreCheck(t *testing.T) {
 				errMsg: "provider not found",
 			},
 		},
+		{
+			name: "wrong value in environment variable RESOURCES_LIMITS_CPU",
+			prepare: func(t *testing.T) {
+				t.Setenv("RESOURCES_LIMITS_CPU", "abcde")
```
Comment (zzxwill) [inline controllers/configuration_controller_test.go:755]: Smart!

**Review by zzxwill — APPROVED** [review]
Good job!

**Review by wonderflow — APPROVED** [review]
great

**Comment by zzxwill** [comment]
@redsnapper2006 Please help resolve conflicting files.

**Comment by zzxwill** [comment]
@redsnapper2006 Please run `make reviewable` to fix the CI issues.

File: controllers/configuration_controller.go (hunk 1 above)
Comment (zzxwill) [reply controllers/configuration_controller.go:224]: 这个是还需要的，对吧？：）

Comment (zzxwill) [reply controllers/configuration_controller.go:224]: 设置了，用户就可以在 helm install 时配置 requests and limits。

Comment (zzxwill) [reply controllers/configuration_controller.go:224]: @redsnapper2006 One last thing to complete:)

Comment (redsnapper2006) [reply controllers/configuration_controller.go:224]: @zzxwill DONE, please confirm

**Review by zzxwill — APPROVED** [review]
Good job!

**Pull request merged**